From: Jonathan Brassow <jbrassow@redhat.com>
Date: Tue, 12 Aug 2008 13:46:16 -0500
Subject: [md] dm-log: move register functions
Message-id: 1218566776.22247.29.camel@hydrogen
O-Subject: [RHEL 5.3 Patch 5 of 8]: device-mapper interface exposure (bz 429337)
Bugzilla: 429337
RH-Acked-by: Alasdair G Kergon <agk@redhat.com>

Patch: dm-log-move-register-functions.patch

 brassow

Reorder a couple of functions in the file so the next patch is readable.

Originator: Alasdair G Kergon <agk@redhat.com>
Signed-off-by: Jonathan Brassow <jbrassow@redhat.com>

diff --git a/drivers/md/dm-log.c b/drivers/md/dm-log.c
index d4153ed..0cd48e5 100644
--- a/drivers/md/dm-log.c
+++ b/drivers/md/dm-log.c
@@ -18,33 +18,6 @@
 static LIST_HEAD(_log_types);
 static DEFINE_SPINLOCK(_lock);
 
-int dm_dirty_log_type_register(struct dm_dirty_log_type *type)
-{
-	spin_lock(&_lock);
-	type->use_count = 0;
-	list_add(&type->list, &_log_types);
-	spin_unlock(&_lock);
-
-	return 0;
-}
-EXPORT_SYMBOL(dm_dirty_log_type_register);
-
-int dm_dirty_log_type_unregister(struct dm_dirty_log_type *type)
-{
-	spin_lock(&_lock);
-
-	if (type->use_count)
-		DMWARN("Unregister failed: log type '%s' still in use",
-		       type->name);
-	else
-		list_del(&type->list);
-
-	spin_unlock(&_lock);
-
-	return 0;
-}
-EXPORT_SYMBOL(dm_dirty_log_type_unregister);
-
 static struct dm_dirty_log_type *_get_type(const char *type_name)
 {
 	struct dm_dirty_log_type *type;
@@ -129,6 +102,32 @@ static void put_type(struct dm_dirty_log_type *type)
 	spin_unlock(&_lock);
 }
 
+int dm_dirty_log_type_register(struct dm_dirty_log_type *type)
+{
+	spin_lock(&_lock);
+	type->use_count = 0;
+	list_add(&type->list, &_log_types);
+	spin_unlock(&_lock);
+
+	return 0;
+}
+EXPORT_SYMBOL(dm_dirty_log_type_register);
+
+int dm_dirty_log_type_unregister(struct dm_dirty_log_type *type)
+{
+	spin_lock(&_lock);
+
+	if (type->use_count)
+		DMWARN("Attempt to unregister a log type that is still in use");
+	else
+		list_del(&type->list);
+
+	spin_unlock(&_lock);
+
+	return 0;
+}
+EXPORT_SYMBOL(dm_dirty_log_type_unregister);
+
 struct dm_dirty_log *dm_dirty_log_create(const char *type_name,
                                         struct dm_target *ti,
 				      unsigned int argc, char **argv)
